presentation/matrix_view.py

```diff
@@ -1,33 +1,46 @@
 import itertools
-from datetime import datetime
-from typing import Dict, List, Optional
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple
 
 import dash_core_components as dcc
 import dash_html_components as html
 
-from domain.config import Config
+from domain.config import Config, MatrixCellColor
 from domain.config.thresholds import Thresholds
 from domain.geo import calc_distance_in_kilometers
 from domain.metric_type import MetricType
-from domain.model import MeshResults
+from domain.model import Agents, MeshResults
 from domain.model.mesh_results import MeshColumn
-from domain.types import MetricValue, Threshold
+from domain.types import AgentID, MetricValue, Threshold
 from presentation.localtime import utc_to_localtime
 
 
+# SLALevel represents measured value in regard to threshold, as a value in connection matrix cell
```

Comment:
I don't understand the comment.

Reply:
I build changes one on top of another so that while merging to main you don't encounter merge conflicts (they touch the same files, main.py, matrix_view.py).
As for the comment:
The levels represent how a measured value eg Latency relates to configured threshold, eg Warning, and they are inputted in the connection matrix displayed to the user. It may be not quite clear; maybe you can help rephrasing the sentence?

Reply:
Coloring used to have a flaw: didn't handle the None values on matrix diagonal

Reply:
I understand that changes are built on top of each other and that is not a problem at all.

However, I still don't understand what was the problem with coloring of matrix cells and (as a consequence) how the code change solves it.

Can you, please, give me an example of how the newly introduced "levels" affect coloring of cells in the matrix? I personally don't see any need for any additional mapping. What am I missing?

Reply:
The below check was supposed to detect whether all cells are assigned the same value (GOOD, green color):
        if all([i == 0.0 for i in itertools.chain(*z)]):
            return [[0.0, GREEN], [1.0, GREEN]]
But it was not handling the fact that the matrix diagonal is filled with None, so wrong branch was taken and wrong color scale was used (which looked OK on Chrome, but not on Firefox - darker shade of green than in matrix legend).
"levels" is introduced to stop setting and checking that the cell is assigned 0.0, 0.5, 1.0 magical values.

Reply:
Please don't focus too much on color scales here; this code will change when I implement "no data" case handling with configurable cell color. Which will happen soon.